app/src/test/kotlin/com/intive/picover/profile/delete/viewmodel/DeleteAccountViewModelTest.kt

```diff
@@ -0,0 +1,47 @@
+package com.intive.picover.profile.delete.viewmodel
+
+import com.intive.picover.auth.model.AccountDeletionResult
+import com.intive.picover.auth.repository.AuthRepository
+import com.intive.picover.common.coroutines.CoroutineTestExtension
+import com.intive.picover.profile.delete.model.DeleteAccountEvent
+import com.intive.picover.profile.delete.model.DeleteAccountSideEffect
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.inspectors.forAll
+import io.kotest.matchers.shouldBe
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class DeleteAccountViewModelTest : ShouldSpec(
+	{
+		extension(CoroutineTestExtension())
+
+		val authRepository: AuthRepository = mockk()
+		val tested by lazy { DeleteAccountViewModel(authRepository) }
+
+		should("emit effect depending on result WHEN DeleteAccount event emitted") {
+
+			listOf(
+				AccountDeletionResult.Success to DeleteAccountSideEffect.ShowSuccessMessage,
+				AccountDeletionResult.ReAuthenticationNeeded to DeleteAccountSideEffect.ShowReAuthenticationNeededMessage,
+			).forAll { (deletionResult, sideEffect) ->
+				runTest(UnconfinedTestDispatcher()) {
+					lateinit var result: DeleteAccountSideEffect
+					coEvery { authRepository.deleteAccount() } returns deletionResult
+					backgroundScope.launch {
```

Comment:
That's needed to collect the sideEffects. I didn't find an easier way to do that

Reply:
@BorysChajdas Can you please resolve the conversation?